File: python/domain_tags.py
import sdk.python.entities
from sdk.python.domains import Domain
from sdk.python.entities import Property


class DomainTag(sdk.python.entities.Entity):

    json_name = 'domainTag'
    resource = '/domains_tags/'

    id = Property(int)
    name = Property(str)
    # colour is in RRGGBB format.
    colour = Property(int)
    description = Property(str)

    domain = Property(Domain, backref="tags")


class DomainTags(sdk.python.entities.Resource):

    entity_class = DomainTag
    json_name = 'domainTags'


domain_tags = DomainTags()

File: php/src/domain_tags.php
<?php
declare(strict_types=1);

require_once 'entity.php';
require_once 'domains.php';
require_once 'users.php';

class DomainTag extends Entity
{
    public static $resource = '/domain_tags/';
    public static $json_name = 'domain_tag';

    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('colour', 'integer');
        $this->json_property('name', 'string');
        $this->json_property('description', 'string');
        $this->json_property('show_public', 'boolean');
        $this->json_property('domain', 'Domain', '',
            $many = False, $recursive = True);
        $this->json_property('jobs', 'Job', '',
            $many = True, $recursive = True);
        $this->json_property('products', 'Product', '',
            $many = True, $recursive = True);
        $this->json_property('invoices', 'Job', '',
            $many = True, $recursive = True);
        $this->json_property('shipments', 'Shipment', '',
            $many = True, $recursive = True);
    }
}


